test/language/expressions/assignment/destructuring/array-elem-put-constant-no-strict.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    The assignment target should obey `const` semantics outside of strict mode.
+es6id: 12.14.5.3
+features: [const]
+flags: [noStrict]
+---*/
+
+var value = [1];
+const c = null;
+var result;
+
+result = [ c ] = value;
```

Comment:
> Or do you mean something else?

Nope--I'm was just hung up on the name GlobalDeclarationInstantiation. As you point out, every abstract operation that creates an execution environment instantiates constants in this way.

> Yes, that seems reasonable.

Great; I'll start on this now